File: <100/026/026.cpp
#include<iostream>
#include<set>
using namespace std;
#define MAX 1000

int answer, period;
const int max_loop = MAX, max_d = MAX;

int get_period(int n){
  int r = 10;
  set<int> s; s.insert(r);

  int num_loop = 0;
  while(num_loop < max_loop){
    ++num_loop;
    bool flag = r%n == 0;
    r = (r%n) * 10;

    if(flag || s.count(r) == 1){
      return num_loop;
    }

    s.insert(r);
  }

  return -1;
}


int main(){
  for(int n = 2; n < max_d; ++n){
    int n_period = get_period(n);
    if(period < n_period){
      period = n_period;
      answer = n;
    }
  }

  cout << answer << ' ' << period << endl;
  return 0;
}

File: <100/026/026.py
import time


def get_period(n: int) -> int:
    r = 10
    s = set([r])
    num_loop = 0
    max_loop = 1000
    while num_loop < max_loop:
        num_loop += 1
        r = (r%n) * 10

        if r == 0 or r in s:
            return num_loop
        
        s.add(r)
    
    return -1


def main() -> int:
    upper_limit: int = 1000
    periods = map(lambda n: (get_period(n), n), range(2,upper_limit))
    answer  = max(periods)[1]
    
    return answer


if __name__ == "__main__":
    start = time.time()
    answer = main()
    elapsed = time.time() - start

    print(f'{elapsed} sec')
    print(answer)
